src/gasPriceOracle/oracle.ts

```diff
@@ -60,39 +88,42 @@ function getEthersGasPriceEstimate(
   assert(gasPriceFeed || legacyFallback, `No suitable gas price oracle for Chain ID ${chainId}`);
   gasPriceFeed ??= chainIsOPStack(chainId) ? ethereum.eip1559 : ethereum.legacy;
 
-  return gasPriceFeed(provider, chainId);
+  return gasPriceFeed(provider, opts);
 }
 
 /**
  * Provide an estimate for the current gas price for a particular chain.
  * @param providerOrChainId A valid ethers provider or a chain ID.
- * @param transport An optional transport object for custom gas price retrieval.
- * @returns Am object of type GasPriceEstimate.
+ * @param transport An optional Viem Transport object for custom gas price retrieval.
+ * @param unsignedTx Only used in Linea provider to estimate priority gas fee.
+ * @returns An object of type GasPriceEstimate.
  */
-export async function getViemGasPriceEstimate(
+export async function _getViemGasPriceEstimate(
   providerOrChainId: providers.Provider | number,
-  transport?: Transport
+  opts: GasPriceEstimateOptions
 ): Promise<GasPriceEstimate> {
+  const { baseFeeMultiplier, transport } = opts;
+
   const chainId =
     typeof providerOrChainId === "number" ? providerOrChainId : (await providerOrChainId.getNetwork()).chainId;
   const viemProvider = getPublicClient(chainId, transport);
 
-  const gasPriceFeeds = {
-    [CHAIN_IDs.ALEPH_ZERO]: arbitrumViem.eip1559,
-    [CHAIN_IDs.ARBITRUM]: arbitrumViem.eip1559,
+  const gasPriceFeeds: Record<
+    number,
+    (provider: PublicClient, opts: GasPriceEstimateOptions) => Promise<InternalGasPriceEstimate>
+  > = {
```

Comment:
ooc, do we need to be explicit on this typing? It should have previously been inferred OK.

Reply:
you mean removing the `as const`? We can do that yes 20461a6ae6dea975870334b34dd66e1af99d8abb

Reply:
Ah, actually I meant the type definitions that were added here. Were they necessary due to some tsc issue? It seems like they should be redundant as long as the members of the object are consistent.

```
 Record<
    number,
    (provider: PublicClient, opts: GasPriceEstimateOptions) => Promise<InternalGasPriceEstimate>
  > = {
```

Reply:
oh true a232f056d7261f428c75208c167e1889cc66dbb1